### PR TITLE
Refactors WKWebview example

### PR DESCRIPTION
**Pull request by Gazzini**
Thanks for putting up this example -- it's super helpful!

There were a few confusing points though, so I tried to clear it up a bit. Specifically:

- Removes an apparently un-used containerview reference
- Breaks up large methods
- Syntactic cleanup

I suspect that these changes will make it easier for future devs to quickly understand this code -- thanks again for making it!

### CODE REVIEW

File: examples/wkwebview/wkwebview/LinkViewController.swift [hunk 1]
```diff
@@ -69,62 +71,81 @@ class LinkViewController: UIViewController, WKNavigationDelegate {
     }
     
     func webView(_ webView: WKWebView,
-                 decidePolicyForNavigationAction navigationAction: WKNavigationAction,
-                 decisionHandler: ((WKNavigationActionPolicy) -> Void)) {
+                 decidePolicyFor navigationAction: WKNavigationAction,
+                 decisionHandler: @escaping ((WKNavigationActionPolicy) -> Void)) {
```
Comment (Gazzini) [inline examples/wkwebview/wkwebview/LinkViewController.swift]: I struggled with this decision -- I eventually decided to just convert it to Swift3, no worries if you disagree :smile:

File: examples/wkwebview/wkwebview/LinkViewController.swift [hunk 2]
```diff
@@ -10,42 +10,44 @@ import WebKit
...
+        webView.allowsBackForwardNavigationGestures = false
+        view = webView
+    }
+    
+    private func loadLinkModuleInWebview() {
         let linkUrl = generateLinkInitializationURL()
         let url = NSURL(string: linkUrl)
         let request = NSURLRequest(url:url! as URL)
-        self.webView.load(request as URLRequest)
-        self.webView.allowsBackForwardNavigationGestures = false
```
Comment (Gazzini) [inline examples/wkwebview/wkwebview/LinkViewController.swift]: I moved it to the `createWebview()` method -- can move it back here if you want, let me know.

File: examples/wkwebview/wkwebview/LinkViewController.swift (hunk 1 above)
Comment (Gazzini) [inline examples/wkwebview/wkwebview/LinkViewController.swift]: sure -- I guess the point is to clearly indicate functionality above all else. makes sense.

**Comment by Gazzini** [comment]
Made a first attempt at addressing feedback -- back to you!

**Comment by michaelckelly** [comment]
@Gazzini - Thanks again for this PR. We're eager to get this merged!

@pbernasconi  - What do you think about merging this?

File: examples/wkwebview/wkwebview/LinkViewController.swift [hunk 3]
```diff
@@ -10,42 +10,44 @@ import WebKit
...
     
-    @IBOutlet var containerView : UIView? = nil
     
     var webView: WKWebView!
     
     override func loadView() {
-        self.webView = WKWebView()
-        self.webView.navigationDelegate = self
-        self.view = webView
+        createWebview()
```
Comment (afh) [inline examples/wkwebview/wkwebview/LinkViewController.swift]: I'd like to suggest making `createWebview` more functional by having it return the `WKWebView` that is then assigned to the `view` property and by adding `delegate` parameter to `createWebview`, which is assigned
to the `navigationDelegate` property of the returned `WKWebView`, e.g.:
`view = createWebView(self)`

This sort of code is more explicit and in line with the documentation:
> […] assign the root view of your view hierarchy to the view property.
[UIViewController.loadView()](https://developer.apple.com/reference/uikit/uiviewcontroller/1621454-loadview)

Out of curiosity what is the motivation for the `createWebview()` method?

**Comment by Gazzini** [comment]
@afh Agreed that returning the view & passing the delegate as a param does make the code more obvious (I also made sure `delegate` was a named param, so it looks like: `view = createWebView(delegate: self)`.

As for the motivation of `createWebView()`: When I was first reading through the example, I was surprised that you were overriding `loadView()`. My first question was "What are they doing here?", and the answer was that you were creating a webview. I thought that if I called a function named "createWebView()", other people might come to that conclusion a few seconds sooner than they would otherwise.

If I was doing it myself, I might have just made the webView a subView of the main UIView, instead of overriding loadView. In my subjective opinion, it's easier for most people to understand, and it's easy to make minor adjustments (like lowering the webView to not overlap with the statusbar). But, there are advantages to overriding `loadView()` as well.

**Comment by afh** [comment]
@Gazzini Thanks for making the modifications and thank you for clarifying the intention.

I agree with you that overriding `loadView()` instead of using `viewDidLoad()` to insert a `WKWebView` into the view hierarchy is uncommon and yet both approaches as you say have their pros and cons.

Furthermore you write:
> My first question was "What are they doing here?", and the answer was that you were creating a webview. I thought that if I called a function named "createWebView()", other people might come to that conclusion a few seconds sooner than they would otherwise.

So, a source code comment explaining why `loadView()` was chosen over `viewDidLoad()` would've been helpful?

**Comment by Gazzini** [comment]
Yes, a comment would be helpful.

**Pull request closed without merging**